eth/abc.py

```diff
@@ -72,6 +73,58 @@ class MiningHeaderAPI(rlp.Serializable, ABC):
     timestamp: int
     extra_data: bytes
 
+    @property
+    @abstractmethod
+    def hash(self) -> Hash32:
+        """
+        Return the hash of the block header.
+        """
+        ...
+
+    @property
+    @abstractmethod
+    def mining_hash(self) -> Hash32:
+        """
+        Return the mining hash of the block header.
+        """
+
+    @property
+    def hex_hash(self) -> str:
+        """
+        Return the hash as a hex string.
+        """
+        ...
+
+    @property
+    @abstractmethod
+    def is_genesis(self) -> bool:
+        """
+        Return ``True`` if this header represents the genesis block of the chain,
+        otherwise ``False``.
+        """
+        ...
+
+    @abstractmethod
+    def copy(self, *args: Any, **kwargs: Any) -> THeader:
+        """
+        Return a copy of the header, optionally overwriting any of its properties.
+        """
+        ...
+
+    @abstractmethod
+    def build_changeset(self, *args: Any, **kwargs: Any) -> Any:
+        """
+        Open a changeset to modify the header.
+        """
+        ...
+
+    @abstractmethod
+    def as_dict(self) -> Dict[Hashable, Any]:
```

Comment:
Maybe add a quick comment on each of the methods that are copied over from `rlp.Serializeable`, saying that they can be deleted from this API after `pyrlp` properly exposes the types? (I think just these last three methods)

*Edit: and the methods in the other classes, of course.*

Reply:
Will do

Reply:
I added comments on all methods. Btw, I've also tried grouping all of them in a `RLPSerializableAPI(Generic[T])` so that we don't have to manually add all these methods to all these different types and while this worked great for >= py37 it yielded an obscure meta class MRO error that I couldn't resolve.